test/modules/TestResultCoordinator/Reports/CountingReportGenerator.cs

```diff
@@ -207,11 +225,26 @@ public async Task<ITestResultReport> CreateReportAsync()
                 totalMatchCount,
                 totalDuplicateExpectedResultCount,
                 totalDuplicateActualResultCount,
+                totalMisorderedActualResultCount,
                 new List<TestOperationResult>(unmatchedResults).AsReadOnly(),
                 eventHubSpecificReportComponents,
                 lastLoadedResultCreatedAt);
         }
 
+        bool IsActualResultSequenceNumberOlder(TestOperationResult actualResult, TestOperationResult expectedResult)
+        {
+            // TODO: The controller for TestResultCoordinator takes in a custom type
+            // not derived from the original types the test modules send. This
+            // means we have to rely on string magic like this to get the sequence
+            // numbers. In order to clean this up we should allow the controller to
+            // ingest the original type the test modules are sending. Then we
+            // can cast to MessageTestResult and grab the sequence number attribute.
+            int actualSequenceNumber = int.Parse(actualResult.Result.Split(";")[2]);
+            int expectedSequenceNumber = int.Parse(expectedResult.Result.Split(";")[2]);
```

Comment:
We could theoretically clean it up in this PR. The fix would be to align the type that the test modules send with the TestOperationResult type that the TRC ingests. It would be nice to do this, however I believe we would need to implement some sort of manual serialization / deserialization which I think would take more time than the reward we get from doing it.

If anyone wants to discuss this further I think it would be a good idea to chat in person. My stance is that with the outstanding work / debt we have already accrued, it is not worth cleaning up this slightly messy string parse.